### ISSUE TITLE
2016.3.3 highstate output with verbose gives new strange format

### ISSUE THREAD

**Issue by Sakorah** [saltstack/salt#35989]
Hi!

I just upgraded to 2016.3.3 and ran a state.highstate on my master (and also on a local vm with salt-call --local for testing). Master and Minions are on 2016.3.3. Further I use state_output=changes and state_verbose=True for my output formatting within the highstate.

With 2016.3.2 and earlier the output worked just like the documentation states:
By default state_verbose is set to True, setting this to False will instruct the highstate outputter to omit displaying anything in green, this means that nothing with a result of True and no changes will not be printed
If changes is used, then terse output will be used if there was no error and no changes, otherwise full output will be used.

So for every state which returns True I get one line and for everything else I get the full output.

With 2016.3.3 and the same settings I get a kinda new, strange and confusing output formatting like this:
```
    service_|-ssh-service_|-ssh_|-running:
        ----------
        __id__:
            ssh-service
        __run_num__:
            9
        changes:
            ----------
        comment:
            The service ssh is already running
        duration:
            25.735
        name:
            ssh
        result:
            True
        start_time:
            15:10:52.798773
```
If I turn of state_verbose and set it to False the formatting gets back to normal but I loose everything which returns True in the output.
```
Salt Version:
           Salt: 2016.3.3

Dependency Versions:
           cffi: 0.9.2
       cherrypy: 3.2.2
       dateutil: 1.5
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: 0.22.2
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 0.9.1
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: 1.2.3
      pycparser: 2.10
       pycrypto: 2.6.1
         pygit2: 0.22.0
         Python: 2.7.6 (default, Jun 22 2015, 17:58:13)
   python-gnupg: Not Installed
         PyYAML: 3.10
          PyZMQ: 14.0.1
           RAET: Not Installed
          smmap: 0.8.2
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: Ubuntu 14.04 trusty
        machine: x86_64
        release: 3.13.0-95-generic
         system: Linux
        version: Ubuntu 14.04 trusty
```

**Comment by Ch3LL** [comment]
@Sakorah i'm having a hard time replicating this on 2016.3.3:

```
root@vagrant-ubuntu-trusty-64:~# cat /srv/salt/test.sls 
succeed:
  cmd.run:
    - name: echo 0

fail:
  cmd.run:
    - name: exit 1

add_file:
  file.managed:
    - name: /tmp/blah
add_file2:
  file.managed:
    - name: /tmp/blah2

add_servcie:
  service.running:
    - name: firewalld
```

```
root@vagrant-ubuntu-trusty-64:~# grep -E -v '^$|#' /etc/salt/master
state_output: changes
state_verbose: True
```

```
root@vagrant-ubuntu-trusty-64:~# salt '*' state.highstate                                                                                                                    
vagrant-ubuntu-trusty-64:
----------
          ID: succeed
    Function: cmd.run
        Name: echo 0
      Result: True
     Comment: Command "echo 0" run
     Started: 15:32:03.581997
    Duration: 13.707 ms
     Changes:   
              ----------
              pid:
                  9806
              retcode:
                  0
              stderr:
              stdout:
                  0
----------
          ID: fail
    Function: cmd.run
        Name: exit 1
      Result: False
     Comment: Command "exit 1" run
     Started: 15:32:03.595892
    Duration: 8.177 ms
     Changes:   
              ----------
              pid:
                  9807
              retcode:
                  1
              stderr:
              stdout:
  Name: /tmp/blah - Function: file.managed - Result: Clean Started: - 15:32:03.606603 Duration: 10.802 ms
  Name: /tmp/blah2 - Function: file.managed - Result: Clean Started: - 15:32:03.617530 Duration: 5.657 ms
----------
          ID: add_servcie
    Function: service.running
        Name: firewalld
      Result: False
     Comment: The named service firewalld is not available
     Started: 15:32:03.949915
    Duration: 25.053 ms
     Changes:   

Summary for vagrant-ubuntu-trusty-64
------------
Succeeded: 3 (changed=2)
Failed:    2
------------
Total states run:     5
Total run time:  63.396 ms
ERROR: Minions returned with non-zero exit code
```

```
root@vagrant-ubuntu-trusty-64:~# salt --version
salt 2016.3.3 (Boron)
```

Would you mind sharing a sanitized version of you master configuration?

**Comment by Sakorah** [comment]
@Ch3LL I fired up my local vm with vagrant and tested it with my own states and also with your test.sls. 

If I call the test.sls it just gives the right output:
```
root@minion-em-elk01:/srv# salt-call --local --state-verbose=True --state-output=changes state.sls test
[WARNING ] /usr/lib/python2.7/dist-packages/salt/grains/core.py:1493: DeprecationWarning: The "osmajorrelease" will be a type of an integer.

[ERROR   ] Command 'exit 1' failed with return code: 1
[ERROR   ] retcode: 1
[ERROR   ] {'pid': 2924, 'retcode': 1, 'stderr': '', 'stdout': ''}
[WARNING ] State for file: /tmp/blah - Neither 'source' nor 'contents' nor 'contents_pillar' nor 'contents_grains' was defined, yet 'replace' was set to 'True'. As there is no source to replace the file with, 'replace' has been set to 'False' to avoid reading the file unnecessarily.
[WARNING ] State for file: /tmp/blah2 - Neither 'source' nor 'contents' nor 'contents_pillar' nor 'contents_grains' was defined, yet 'replace' was set to 'True'. As there is no source to replace the file with, 'replace' has been set to 'False' to avoid reading the file unnecessarily.
[ERROR   ] The named service firewalld is not available
local:
----------
          ID: succeed
    Function: cmd.run
        Name: echo 0
      Result: True
     Comment: Command "echo 0" run
     Started: 08:45:19.972506
    Duration: 6.625 ms
     Changes:
              ----------
              pid:
                  2923
              retcode:
                  0
              stderr:
              stdout:
                  0
----------
          ID: fail
    Function: cmd.run
        Name: exit 1
      Result: False
     Comment: Command "exit 1" run
     Started: 08:45:19.979332
    Duration: 6.938 ms
     Changes:
              ----------
              pid:
                  2924
              retcode:
                  1
              stderr:
              stdout:
  Name: /tmp/blah - Function: file.managed - Result: Clean Started: - 08:45:19.988246 Duration: 3.101 ms
  Name: /tmp/blah2 - Function: file.managed - Result: Clean Started: - 08:45:19.991471 Duration: 0.821 ms
----------
          ID: add_servcie
    Function: service.running
        Name: firewalld
      Result: False
     Comment: The named service firewalld is not available
     Started: 08:45:20.253565
    Duration: 24.372 ms
     Changes:

Summary for local
------------
Succeeded: 3 (changed=2)
Failed:    2
------------
Total states run:     5
Total run time:  41.857 ms
root@minion-em-elk01:/srv#
```

I tried calling a highstate and some SLS from my environment and got different results:
* Calling my highstate gives the strange output
* Calling some SLS states give the strange output
* Calling some SLS states gives normal output

I don't know exactly what triggers this so I tried isolating some and simplifying one. I created this test.sls where I added just one part of my user.sls creating the user:
```
succeed:
  cmd.run:
    - name: echo 0

fail:
  cmd.run:
    - name: exit 1

add_file:
  file.managed:
    - name: /tmp/blah
add_file2:
  file.managed:
    - name: /tmp/blah2

add_servcie:
  service.running:
    - name: firewalld


#static users
{% for user, args in salt['pillar.get']('user:enabled',{}).items() %}
{{user}}:
  group.present:
    - name: {{user}}

  user.present:
    - name: {{user}}
    - gid_from_name: True
    - home: {% if args.get('home') %} {{ args.get('home') }} {% else %} /home/{{user}} {% endif %}
    - createhome: True
    - password: {{ salt['random.shadow_hash'] }}
    - fullname: {{args.get('fullname')}}
    - enforce_password: False
    - shell: /bin/bash
    - require:
      - group: {{user}}
    - onchanges_in:
      - cmd: user-permission-cmd

{% endfor %}

user-permission-cmd:
  cmd.run:
    - name: chmod 700 /home/*
```

And I have this pillars for my users to be created:
```
root@minion-em-elk01:/srv# salt-call --local --state-verbose=True --state-output=changes pillar.item user:enabled
[WARNING ] /usr/lib/python2.7/dist-packages/salt/grains/core.py:1493: DeprecationWarning: The "osmajorrelease" will be a type of an integer.

local:
    ----------
    user:enabled:
        ----------
        ong:
            ----------
            fullname:
                ong user
```

With this I get the following output:
```
root@minion-em-elk01:/srv# salt-call --local --state-verbose=True --state-output=changes state.sls test test=true
[WARNING ] /usr/lib/python2.7/dist-packages/salt/grains/core.py:1493: DeprecationWarning: The "osmajorrelease" will be a type of an integer.

[WARNING ] State for file: /tmp/blah - Neither 'source' nor 'contents' nor 'contents_pillar' nor 'contents_grains' was defined, yet 'replace' was set to 'True'. As there is no source to replace the file with, 'replace' has been set to 'False' to avoid reading the file unnecessarily.
[WARNING ] State for file: /tmp/blah2 - Neither 'source' nor 'contents' nor 'contents_pillar' nor 'contents_grains' was defined, yet 'replace' was set to 'True'. As there is no source to replace the file with, 'replace' has been set to 'False' to avoid reading the file unnecessarily.
[ERROR   ] The named service firewalld is not available
local:
    ----------
    cmd_|-fail_|-exit 1_|-run:
        ----------
        __id__:
            fail
        __run_num__:
            1
        changes:
            ----------
        comment:
            Command "exit 1" would have been executed
        duration:
            0.332 ms
        name:
            exit 1
        result:
            None
        start_time:
            08:56:54.260690
    cmd_|-succeed_|-echo 0_|-run:
        ----------
        __id__:
            succeed
        __run_num__:
            0
        changes:
            ----------
        comment:
            Command "echo 0" would have been executed
        duration:
            0.489 ms
        name:
            echo 0
        result:
            None
        start_time:
            08:56:54.260091
    cmd_|-user-permission-cmd_|-chmod 700 /home/*_|-run:
        ----------
        __run_num__:
            7
        __sls__:
            test
        changes:
            ----------
        comment:
            State was not run because none of the onchanges reqs changed
        result:
            True
    file_|-add_file2_|-/tmp/blah2_|-managed:
        ----------
        __id__:
            add_file2
        __run_num__:
            3
        changes:
            ----------
        comment:
            File /tmp/blah2 not updated
        duration:
            0.883
        name:
            /tmp/blah2
        pchanges:
            ----------
        result:
            True
        start_time:
            08:56:54.264627
    file_|-add_file_|-/tmp/blah_|-managed:
        ----------
        __id__:
            add_file
        __run_num__:
            2
        changes:
            ----------
        comment:
            File /tmp/blah not updated
        duration:
            2.601
        name:
            /tmp/blah
        pchanges:
            ----------
        result:
            True
        start_time:
            08:56:54.261903
    group_|-ong_|-ong_|-present:
        ----------
        __id__:
            ong
        __run_num__:
            5
        changes:
            ----------
        comment:
            Group ong is present and up to date
        duration:
            0.511
        name:
            ong
        result:
            True
        start_time:
            08:56:54.558868
    service_|-add_servcie_|-firewalld_|-running:
        ----------
        __id__:
            add_servcie
        __run_num__:
            4
        changes:
            ----------
        comment:
            The named service firewalld is not available
        duration:
            26.787 ms
        name:
            firewalld
        result:
            False
        start_time:
            08:56:54.531406
    user_|-ong_|-ong_|-present:
        ----------
        __id__:
            ong
        __run_num__:
            6
        changes:
            ----------
        comment:
            User ong is present and up to date
        duration:
            3.629
        name:
            ong
        result:
            True
        start_time:
            08:56:54.560335
```

**Comment by omolto** [comment]
Same here!! :(

The same sls on an environment with 2016.3.2 outputs the correct format, and on the new 2016.3.3 outputs this strange cryptic format.

You can test in any environment with: salt minion-id state.highstate --state-output=changes

This is a bug introduced in 2016.3.3 version, could you please fix it?

Thanks!

**Comment by romanovna** [comment]
Same for me too. Also I've been noticed, if you try to apply a state twice a row - first try will be ok, but  second will return "fully verbosed" output, but for me - it was reproduced only in 'hard' states, not 'hello-world' like :) 
Master and slaves are 2016-3.3

**Comment by dneades** [comment]
I am seeing this strange and verbose output style with `salt-ssh`, too. (Incidentally, I have been unable to configure the output format of `salt-ssh` at all, even before this latest 2016.3.3 release.)

**Comment by Ch3LL** [comment]
@Sakorah thanks for the test case. I am now able to replicate the behavior. One thing to note is it worked for me when i removed the `onchanges_in` 

The first bad commit is 6cdee210363db16185e32317f85cc49ed58bf29e

We will definitely need to get this fixed up thanks for the heads up guys

**Comment by Ch3LL** [comment]
Also seeing this error in the trace log output: 

```
[DEBUG   ] LazyLoaded config.option
[DEBUG   ] LazyLoaded highstate.output
[DEBUG   ] Traceback (most recent call last):
  File "/home/ch3ll/git/salt/salt/output/__init__.py", line 39, in try_printout
    return get_printout(out, opts)(data).rstrip()
  File "/home/ch3ll/git/salt/salt/output/highstate.py", line 138, in output
    return _format_host(host, hostdata)[0]
  File "/home/ch3ll/git/salt/salt/output/highstate.py", line 286, in _format_host
    msg = _format_terse(tcolor, comps, ret, colors, tabular)
  File "/home/ch3ll/git/salt/salt/output/highstate.py", line 555, in _format_terse
    ret)
KeyError: u'start_time'

[DEBUG   ] LazyLoaded nested.output
```

**Comment by rallytime** [comment]
Hey everyone - please give #36028 a try. That should fix up this issue for you.

**Comment by EvaSDK** [comment]
Applying #36028 fixed `--state-output=changes` for me, thanks

**Comment by moloney** [comment]
This issue also appears in 2015.8.12. Will the fix be backported to the 2015.8.X series?

Since the patch in #36028 doesn't apply cleanly to 2015.8.12 I instead worked around the problem by setting "state_output_profile: False" in the master conf for now.

**Comment by rallytime** [comment]
@moloney Thanks for letting us know. I've opened up #36730 to backport these changes to the 2015.8 branch.

**Comment by EvaSDK** [comment]
Would it be possible to get this in a release, it is a bit annoying to have to patch every new installation.

**Comment by rallytime** [comment]
@EvaSDK This fix is in the upcoming 2016.3.4 release.
